main.go

```diff
@@ -295,6 +302,13 @@ func (o *Options) Validate() error {
 		return fmt.Errorf("--db-only-mode cannot be set to false (deprecated flag soon to be removed, feature now mandatory)")
 	}
 
+	if o.CRTimeRoundingFactor > maxCRTimeRoundingFactor {
+		return fmt.Errorf("please provide a value smaller than %v for --component-readiness-time-rounding-factor", maxCRTimeRoundingFactor)
+	}
+	if o.CRTimeRoundingFactor > 0 {
+		api.CacheDuration = o.CRTimeRoundingFactor
```

Comment:
This isn't enough, you have to make the cache expire at the boundary itself.  If I hit a URL at 12:23 you'll be caching the data until 13:23.  You instead want it to expire at 13:00.

The caching will need to be adjusted for each request, the Cache interface likely needs changing to support this kind of per-request TTL.